src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -260,6 +262,113 @@ void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
   listener_->OnCallstackSample(std::move(sample));
 }
 
+void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
+                                    const SchedWakeupWithStackPerfEventData& event_data) {
+  // TODO(b/243515760) Refactor this function because a lot of this code is duplicated and it can be
+  // written better.
```

Comment:
What are the different parts? Just
```
  TracepointCallstack tracepoint_callstack;
  tracepoint_callstack.set_tid(event_data.woken_tid);
  tracepoint_callstack.set_timestamp_ns(event_timestamp);
```
instead of 
```
  FullCallstackSample sample;
  sample.set_pid(event_data.pid);
  sample.set_tid(event_data.tid);
  sample.set_timestamp_ns(event_timestamp);
```
Can't be done in this change then? Or what am I missing?